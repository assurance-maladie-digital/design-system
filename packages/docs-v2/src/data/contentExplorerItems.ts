import { TreeviewItem } from '~/components/content-explorer/types';

export const contentExplorerItems: TreeviewItem[] = [
	{
		name: 'public',
		path: 'public/index',
		children: [
			{
				name: 'css',
				path: 'public/css/index',
				children: [
					{
						name: 'base.css',
						path: 'public/css/base.css',
						file: 'css'
					},
					{
						name: 'loader.css',
						path: 'public/css/loader.css',
						file: 'css'
					}
				]
			},
			{
				name: 'js',
				path: 'public/js/index',
				children: [
					{
						name: 'lib',
						path: 'public/js/lib/index',
						children: [
							{
								name: 'matomo.js',
								path: 'public/js/lib/matomo.js',
								file: 'js'
							}
						]
					},
					{
						name: 'config.js',
						path: 'public/js/config.js',
						file: 'js'
					},
					{
						name: 'config.js.dist',
						path: 'public/js/config.js.dist',
						file: 'js'
					},
					{
						name: 'config.js.env',
						path: 'public/js/config.js.env',
						file: 'js'
					}
				]
			},
			{
				name: 'android-chrome-192x192.png',
				file: 'png'
			},
			{
				name: 'android-chrome-512x512.png',
				file: 'png'
			},
			{
				name: 'apple-touch-icon.png',
				file: 'png'
			},
			{
				name: 'favicon-16x16.png',
				file: 'png'
			},
			{
				name: 'favicon-32x32.png',
				file: 'png'
			},
			{
				name: 'favicon.ico',
				file: 'ico'
			},
			{
				name: 'index.html',
				file: 'html'
			},
			{
				name: 'robots.txt',
				file: 'txt'
			},
			{
				name: 'safari-pinned-tab.svg',
				file: 'svg'
			},
			{
				name: 'site.webmanifest',
				file: 'json'
			}
		]
	},
	{
		name: 'src',
		path: 'src',
		children: [
			{
				name: 'assets',
				children: [
					{
						name: 'logo.svg',
						file: 'svg'
					}
				]
			},
			{
				name: 'components',
				children: [
					{
						name: 'layout',
						children: [
							{
								name: 'tests',
								children: [
									{
										name: 'AppFooter.spec.ts',
										file: 'ts'
									},
									{
										name: 'AppHeader.spec.ts',
										file: 'ts'
									},
									{
										name: 'AppToolbar.spec.ts',
										file: 'ts'
									},
									{
										name: 'HeaderMenu.spec.ts',
										file: 'ts'
									}
								]
							},
							{
								name: 'AppFooter.vue',
								file: 'vue'
							},
							{
								name: 'AppHeader.vue',
								file: 'vue'
							},
							{
								name: 'AppToolbar.vue',
								file: 'vue'
							},
							{
								name: 'HeaderMenu.vue',
								file: 'vue'
							},
							{
								name: 'index.ts',
								file: 'ts'
							},
							{
								name: 'README.md',
								file: 'md'
							}
						]
					},
					{
						name: 'Links',
						children: [
							{
								name: 'tests',
								children: [
									{
										name: 'Links.spec.ts',
										file: 'ts'
									}
								]
							},
							{
								name: 'index.ts',
								file: 'ts'
							},
							{
								name: 'Links.vue',
								file: 'vue'
							}
						]
					}
				]
			},
			{
				name: 'constants',
				children: [
					{
						name: 'index.ts',
						file: 'ts'
					}
				]
			},
			{
				name: 'plugins',
				children: [
					{
						name: 'axios.ts',
						file: 'ts'
					},
					{
						name: 'form-builder.ts',
						file: 'ts'
					},
					{
						name: 'vue-dot.ts',
						file: 'ts'
					},
					{
						name: 'vuetify.ts',
						file: 'ts'
					},
					{
						name: 'webfontloader.ts',
						file: 'ts'
					}
				]
			},
			{
				name: 'router',
				children: [
					{
						name: 'index.ts',
						file: 'ts'
					},
					{
						name: 'navigationRedirect.ts',
						file: 'ts'
					},
					{
						name: 'routes.ts',
						file: 'ts'
					}
				]
			},
			{
				name: 'services',
				children: [
					{
						name: 'folders',
						children: [
							{
								name: 'api.ts',
								file: 'ts'
							}
						]
					},
					{
						name: 'dateFormatter.ts',
						file: 'ts'
					}
				]
			},
			{
				name: 'store',
				children: [
					{
						name: 'index.ts',
						file: 'ts'
					},
					{
						name: 'types.d.ts',
						file: 'ts'
					}
				]
			},
			{
				name: 'theme',
				children: [
					{
						name: 'colors.ts',
						file: 'ts'
					},
					{
						name: 'styles.scss',
						file: 'scss'
					}
				]
			},
			{
				name: 'translations',
				children: [
					{
						name: 'fr'
					},
					{
						name: 'index.ts',
						file: 'ts'
					}
				]
			},
			{
				name: 'types',
				children: [
					{
						name: 'index.d.ts',
						file: 'ts'
					}
				]
			},
			{
				name: 'views',
				children: [
					{
						name: 'tests',
						children: [
							{
								name: 'About.spec.ts',
								file: 'ts'
							},
							{
								name: 'Home.spec.ts',
								file: 'ts'
							},
							{
								name: 'Maintenance.spec.ts',
								file: 'ts'
							},
							{
								name: 'NotFound.spec.ts',
								file: 'ts'
							}
						]
					},
					{
						name: 'About.vue',
						file: 'vue'
					},
					{
						name: 'Home.vue',
						file: 'vue'
					},
					{
						name: 'Maintenance.vue',
						file: 'vue'
					},
					{
						name: 'NotFound.vue',
						file: 'vue'
					}
				]
			},
			{
				name: 'App.vue',
				file: 'vue'
			},
			{
				name: 'decorators.ts',
				file: 'ts'
			},
			{
				name: 'global.d.ts',
				file: 'ts'
			},
			{
				name: 'i18n.ts',
				file: 'ts'
			},
			{
				name: 'main.ts',
				file: 'ts'
			},
			{
				name: 'modules.d.ts',
				file: 'ts'
			}
		]
	},
	{
		name: 'tests',
		path: 'tests',
		children: [
			{
				name: 'e2e',
				children: [
					{
						name: 'plugins',
						children: [
							{
								name: 'index.js',
								file: 'js'
							}
						]
					},
					{
						name: 'specs',
						children: [
							{
								name: 'About.spec.ts',
								file: 'ts'
							},
							{
								name: 'Home.spec.ts',
								file: 'ts'
							},
							{
								name: 'NotFound.spec.ts',
								file: 'ts'
							}
						]
					},
					{
						name: 'support',
						children: [
							{
								name: 'commands.js',
								file: 'js'
							},
							{
								name: 'index.js',
								file: 'js'
							}
						]
					},
					{
						name: 'types',
						children: [
							{
								name: 'index.d.ts',
								file: 'ts'
							}
						]
					},
					{
						name: 'eslintrc.js',
						file: 'js'
					},
					{
						name: 'tsconfig.json',
						file: 'json'
					}
				]
			},
			{
				name: 'unit',
				children: [
					{
						name: 'App.spec.ts',
						file: 'ts'
					},
					{
						name: 'index.ts',
						file: 'ts'
					}
				]
			}
		]
	},
	{
		name: '.browserslistrc',
		path: '_browserslistrc',
		file: 'txt'
	},
	{
		name: '.editorconfig',
		path: '_editorconfig',
		file: 'txt'
	},
	{
		name: '.eslintrc.js',
		path: '_eslintrc.js',
		file: 'js'
	},
	{
		name: '.gitignore',
		path: '_gitignore',
		file: 'txt'
	},
	{
		name: 'babel.config.js',
		path: 'babel.config.js',
		file: 'js'
	},
	{
		name: 'cypress.json',
		path: 'cypress.json',
		file: 'json'
	},
	{
		name: 'jest.config.js',
		path: 'jest.config.js',
		file: 'js'
	},
	{
		name: 'package.json',
		path: 'package.json',
		file: 'json'
	},
	{
		name: 'postcss.config.js',
		path: 'postcss.config.js',
		file: 'js'
	},
	{
		name: 'README.md',
		path: 'README.md',
		file: 'md'
	},
	{
		name: 'tsconfig.json',
		path: 'tsconfig.json',
		file: 'json'
	},
	{
		name: 'vue.config.js',
		path: 'vue.config.js',
		file: 'js'
	},
	{
		name: 'yarn.lock',
		path: 'yarn.lock',
		file: 'txt'
	}
];
