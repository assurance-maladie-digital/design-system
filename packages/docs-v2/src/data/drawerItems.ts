import { DrawerItem } from '~/types/drawer';

import { mdiSpeedometer, mdiViewDashboard, mdiCogs } from '@mdi/js';

export const drawerItems: DrawerItem[] = [
	{
		title: 'DÃ©marrer',
		icon: mdiSpeedometer,
		items: [
			{
				title: 'Introduction',
				to: '/demarrer/introduction'
			},
			{
				title: 'Installation',
				to: '/demarrer/installation'
			},
			{
				title: 'Contribuer',
				to: '/demarrer/contribuer'
			}
		]
	},
	{
		title: 'Composants',
		icon: mdiViewDashboard,
		items: [
			{
				title: 'CopyBtn',
				to: '/composants/copy-btn'
			},
			{
				title: 'CustomIcon',
				to: '/composants/custom-icon'
			},
			{
				title: 'DataList',
				to: '/composants/data-list'
			},
			{
				title: 'DatePicker',
				to: '/composants/date-picker'
			},
			{
				title: 'DialogBox',
				to: '/composants/dialog-box'
			},
			{
				title: 'ErrorPage',
				to: '/composants/error-page'
			},
			{
				title: 'FileUpload',
				to: '/composants/file-upload'
			},
			{
				title: 'FooterWrapper',
				to: '/composants/footer-wrapper'
			},
			{
				title: 'FormBuilder',
				to: '/composants/form-builder'
			},
			{
				title: 'FormField',
				to: '/composants/form-field'
			},
			{
				title: 'FormFieldList',
				to: '/composants/form-field-list'
			},
			{
				title: 'HeaderLoading',
				to: '/composants/header-loading'
			},
			{
				title: 'LangBtn',
				to: '/composants/lang-btn'
			},
			{
				title: 'NotificationBar',
				to: '/composants/notification-bar'
			},
			{
				title: 'PageCard',
				to: '/composants/page-card'
			},
			{
				title: 'PaginatedTable',
				to: '/composants/paginated-table'
			},
			{
				title: 'SubHeader',
				to: '/composants/sub-header'
			},
			{
				title: 'TableToolbar',
				to: '/composants/table-toolbar'
			},
			{
				title: 'UploadWorkflow',
				to: '/composants/upload-workflow'
			}
		]
	},
	{
		title: 'Composants techniques',
		icon: mdiCogs,
		items: [
			{
				title: 'Directives',
				items: [
					{
						title: 'Debounce',
						to: '/directives/debounce'
					}
				]
			},
			{
				title: 'Starter Kit',
				items: [
					{
						title: 'Explorer',
						to: '/starter-kit/explorer'
					}
				]
			}
		]
	}
];
