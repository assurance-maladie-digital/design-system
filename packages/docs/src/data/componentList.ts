import { ComponentSection } from '~/types/components';
import { ComponentStatusEnum } from '~/constants/ComponentStatusEnum';

export const componentList: ComponentSection[] = [
	{
		title: 'Structure',
		components: [
			{
				name: 'HeaderBar',
				description: 'Utilisé pour afficher l’en-tête d’une page ainsi qu’une barre de navigation.',
				href: '/composants/structure/header-bar',
				image: 'components/header-bar.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'SubHeader',
				description: 'Utilisé pour afficher un bloc d’informations sous l’en-tête principale.',
				href: '/composants/structure/sub-header',
				image: 'components/sub-header.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'FooterBar',
				description: 'Utilisé pour afficher un pied de page.',
				href: '/composants/structure/footer-bar',
				image: 'components/footer-bar.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'FooterWrapper',
				description: 'Utilisé avec le composant FooterBtn pour afficher un pied de page.',
				href: '/composants/structure/footer-wrapper',
				image: 'components/footer-wrapper.svg',
				status: ComponentStatusEnum.DEPRECATED
			}
		]
	},
	{
		title: 'Layout',
		components: [
			{
				name: 'PageContainer',
				description: 'Conteneur transparent utilisé pour afficher une page.',
				href: '/composants/layout/page-container',
				image: 'components/page-container.svg',
				status: ComponentStatusEnum.READY
			}
		]
	},
	{
		title: 'Navigation',
		components: [
			{
				name: 'ExternalLinks',
				description: 'Utilisé pour afficher un menu avec une liste vers des liens externes.',
				href: '/composants/navigation/external-links',
				image: 'components/external-links.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'ContextualMenu',
				description: 'Utilisé pour pour afficher un menu avec une liste d’ancres pour la navigation.',
				href: '/composants/navigation/contextual-menu',
				image: 'components/contextual-menu.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'SkipLink',
				description: 'Utilisé pour permettre à l’utilisateur d’accéder plus facilement au contenu principal.',
				href: '/composants/navigation/skip-link',
				image: 'components/skip-link.svg',
				status: ComponentStatusEnum.READY
			}
		]
	},
	{
		title: 'Boutons',
		components: [
			{
				name: 'BackBtn',
				description: 'Utilisé pour permettre à l’utilisateur de revenir à la page précédente.',
				href: '/composants/boutons/back-btn',
				image: 'components/back-btn.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'BackToTopBtn',
				description: 'Utilisé pour afficher un bouton permettant à l’utilisateur de remonter en haut d’une page.',
				href: '/composants/boutons/back-to-top-btn',
				image: 'components/back-to-top-btn.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'CopyBtn',
				description: 'Utilisé pour afficher un bouton permettant à l’utilisateur de copier du texte.',
				href: '/composants/boutons/copy-btn',
				image: 'components/copy-btn.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'LangBtn',
				description: 'Utilisé pour permettre à l’utilisateur de choisir la langue de l’application',
				href: '/composants/boutons/lang-btn',
				image: 'components/lang-btn.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'DownloadBtn',
				description: 'Utilisé pour permettre à l’utilisateur de télécharger un document provenant d’une API.',
				href: '/composants/boutons/download-btn',
				image: 'components/download-btn.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'FranceConnectBtn',
				description: 'Utilisé pour afficher un bouton de connexion à FranceConnect.',
				href: '/composants/boutons/france-connect-btn',
				image: 'components/france-connect-btn.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'UserMenuBtn',
				description: 'Utilisé pour afficher un bouton activant un menu d’actions liées à l’utilisateur.',
				href: '/composants/boutons/user-menu-btn',
				image: 'components/user-menu-btn.svg',
				status: ComponentStatusEnum.READY
			}
		]
	},
	{
		title: 'Formulaires',
		components: [
			{
				name: 'DatePicker',
				description: 'Utilisé pour permettre à l’utilisateur de sélectionner ou de saisir une date.',
				href: '/composants/formulaires/date-picker',
				image: 'components/date-picker.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'FileUpload',
				description: 'Utilisé pour permettre à l’utilisateur de sélectionner ou de glisser-déposer des fichiers.',
				href: '/composants/formulaires/file-upload',
				image: 'components/file-upload.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'NirField',
				description: 'Utilisé pour permettre à l’utilisateur de saisir un numéro de Sécurité sociale (NIR).',
				href: '/composants/formulaires/nir-field',
				image: 'components/nir-field.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'PhoneField',
				description: 'Utilisé pour permettre à l’utilisateur de saisir un numéro de téléphone.',
				href: '/composants/formulaires/phone-field',
				image: 'components/phone-field.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'PeriodField',
				description: 'Utilisé pour permettre à l’utilisateur de saisir une période.',
				href: '/composants/formulaires/period-field',
				image: 'components/period-field.svg',
				status: ComponentStatusEnum.NEW
			},
			{
				name: 'SearchListField',
				description: 'Utilisé pour permettre à l’utilisateur de sélectionner une valeur dans une liste.',
				href: '/composants/formulaires/search-list-field',
				image: 'placeholder.svg',
				status: ComponentStatusEnum.NEW
			},
			{
				name: 'UploadWorkflow',
				description: 'Utilisé pour permettre à l’utilisateur de sélectionner une liste de fichiers.',
				href: '/composants/formulaires/upload-workflow',
				image: 'components/upload-workflow.svg',
				status: ComponentStatusEnum.READY
			}
		]
	},
	{
		title: 'Tableaux',
		components: [
			{
				name: 'PaginatedTable',
				description: 'Utilisé pour afficher une VDataTable de Vuetify avec une pagination persistante.',
				href: '/composants/tableaux/paginated-table',
				image: 'components/paginated-table.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'TableToolbar',
				description: 'Utilisé pour afficher une barre au-dessus des tableaux.',
				href: '/composants/tableaux/table-toolbar',
				image: 'components/table-toolbar.svg',
				status: ComponentStatusEnum.READY
			}
		]
	},
	{
		title: 'Filtres',
		components: [
			{
				name: 'FiltersInline',
				description: 'Utilisé pour afficher des filtres directement dans la page.',
				href: '/composants/filtres/filters-inline',
				image: 'components/filters-inline.svg',
				status: ComponentStatusEnum.NEW
			},
			{
				name: 'FiltersSidebar',
				description: 'Utilisé pour afficher des filtres dans un panel dédié.',
				href: '/composants/filtres/filters-side-bar',
				image: 'components/filters-sidebar.svg',
				status: ComponentStatusEnum.NEW
			},
			{
				name: 'FilterModule',
				description: 'Utilisé pour permettre à l’utilisateur de sélectionner des filtres.',
				href: '/composants/filtres/filter-module',
				image: 'components/filter-module.svg',
				status: ComponentStatusEnum.DEPRECATED
			}
		]
	},
	{
		title: 'Données',
		components: [
			{
				name: 'ChipList',
				description: 'Utilisé pour afficher une liste de puces.',
				href: '/composants/donnees/chip-list',
				image: 'placeholder.svg',
				status: ComponentStatusEnum.NEW
			},
			{
				name: 'DataList',
				description: 'Utilisé pour afficher une liste de données.',
				href: '/composants/donnees/data-list',
				image: 'components/data-list.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'DataListGroup',
				description: 'Utilisé pour afficher une liste de DataList.',
				href: '/composants/donnees/data-list-group',
				image: 'components/data-list-group.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'FileList',
				description: 'Utilisé pour afficher une liste de fichiers.',
				href: '/composants/donnees/file-list',
				image: 'components/file-list.svg',
				status: ComponentStatusEnum.NEW
			}
		]
	},
	{
		title: 'Feedback',
		components: [
			{
				name: 'DialogBox',
				description: 'Utilisé pour afficher une boîte de dialogue avec des boutons d’actions.',
				href: '/composants/feedback/dialog-box',
				image: 'components/dialog-box.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'NotificationBar',
				description: 'Utilisé avec le module Vuex notification pour afficher des notifications à l’utilisateur.',
				href: '/composants/feedback/notification-bar',
				image: 'components/notification-bar.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'CookieBanner',
				description: 'Utilisé pour permettre à l’utilisateur d’accepter ou non les cookies.',
				href: '/composants/feedback/cookie-banner',
				image: 'components/cookie-banner.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'HeaderLoading',
				description: 'Utilisé pour afficher un élément de chargement avec des dimensions personnalisées.',
				href: '/composants/feedback/header-loading',
				image: 'components/header-loading.svg',
				status: ComponentStatusEnum.READY
			},
			{
				name: 'RatingPicker',
				description: 'Permet de recueillir l’avis d’un utilisateur.',
				href: '/composants/feedback/rating-picker',
				image: 'components/rating-picker.svg',
				status: ComponentStatusEnum.READY
			}
		]
	}
];
