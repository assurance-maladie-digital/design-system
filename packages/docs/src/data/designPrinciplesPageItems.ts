import { mdiHeartPulse, mdiAccountGroup, mdiAutoFix } from '@mdi/js';

export const designPrinciplesPageItems = [
	{
		title: 'Être au service des usagers de la santé publique',
		description: 'En tant que Design System pour l’Assurance Maladie, notre devoir est de **répondre avec efficacité et intelligence aux besoins** de nos différents profils d’utilisateur citoyen.',
		icon: mdiHeartPulse,
		content: [
			'<u>Nous recommandons</u> pour chaque utilisation de notre Design System de **favoriser l’écoute et l’empathie auprès des utilisateurs finaux** en récoltant dès que le projet le permet un ou plusieurs retours d’expériences. En suivant ce principe, c’est depuis 2017 plus de 20 produits et services numériques qui ont permis d’affiner et de structurer les choix ergonomiques de notre Design System.',
			'Afin d’accompagner les équipes de l’Assurance Maladie, **nous pouvons aider à la mise en place de tests utilisateurs, au passage et à l’analyse des résultats**. En cohérence avec les besoins ou les contraintes des projets, nous vous conseillerons sur la typologie de test à adopter afin de répondre avec performance à vos besoins.'
		]
	},
	{
		title: 'Assurer l’accessibilité pour tous',
		description: '**L’accès aux produits numériques doit être permis à tous**. Pour cela, nous favorisons dans notre démarche de design le respect des [critères d’accessibilité](/fondamentaux/accessibilite) définis par le Référentiel Général d’Amélioration de l’Accessibilité.',
		icon: mdiAccountGroup,
		content: [
			'<u>Nous recommandons</u> pour chaque utilisation de notre Design System de ne pas transformer les composants. La conception des composants vise à respecter quatre indices de performance. **Nos composants et leurs contenus informationnels sont perceptibles (1), manipulables (2), compréhensibles (3) et robustes (4) dans l’usage**. En complément, nous recommandons de proposer de la redondance informationnelle par un ou plusieurs moyens alternatifs lorsqu’une information est complexe à comprendre ou à atteindre sur l’interface.',
			'Afin d’accompagner les équipes de l’Assurance Maladie, **nous pouvons aider à l’élaboration de vos interfaces pour qu’elles respectent les normes d’accessibilité**. Dans le cas où nos composants ne correspondent pas à vos exigences, nous pouvons collaborer avec vous à leurs adaptations.'
		]
	},
	{
		title: 'Rendre évident ce qui est complexe',
		description: 'Malgré la complexité des projets, notre Design System doit permettre **la construction d’interfaces fluides et simples d’utilisation** pour assurer aux utilisateurs finaux une bonne expérience numérique.',
		icon: mdiAutoFix,
		content: [
			'<u>Nous recommandons</u> pour chaque utilisation de notre Design System de favoriser des **compositions graphiques d’interfaces peu chargées avec des espaces vierges et une hiérarchisation des informations**. Afin de permettre le responsive design, les interfaces doivent être structurées en utilisant des grilles et permettre ainsi l’affichage multi-support. En complément, nous recommandons de favoriser des parcours utilisateurs courts et de les concevoir en respectant pour chaque étape [la théorie de l’action proposée par Don Norman](https://fr.wikipedia.org/wiki/Th%C3%A9orie_de_l%27action_(Norman)).',
			'Afin d’accompagner les équipes de l’Assurance Maladie, nous pouvons aider à structurer les interfaces. En proposant par exemple des ateliers de cadrage participatifs dont l’objectif est de relever auprès de vos besoins et des besoins des utilisateurs finaux les informations pertinentes (insights) à afficher. C’est une étape importante pour l’organisation graphique des interfaces.'
		]
	}
];
