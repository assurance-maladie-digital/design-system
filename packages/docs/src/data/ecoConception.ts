export const ecoConception = [
	{
		title: 'Catégorie 1\xa0: un design épuré',
		description: 'Privilégier un design épuré. Un bon design n’a pas besoin d’être chargé pour être agréable, au contraire. <br/><br/>' +
		'Certains composants ne sont souvent pas utiles, et ne servent qu’à habiller une page. <br/>' +
		'Le carrousel est un bon exemple\xa0: c’est un composant lourd, comportant à la fois du Javascript et des images de grandes tailles.' +
		'Il est pourtant rarement indispensable à la navigation et fait souvent doublon avec le menu.'
	},
	{
		title: 'Catégorie 2\xa0: les couleurs',
		description: 'Le fond d’écran noir permet d’économiser de l’énergie sur les écrans OLED. Il est donc important de prévoir le mode sombre sur les nouveaux composants.'
	},
	{
		title: 'Catégorie 3\xa0: les images',
		description: 'Le choix des images est important. <br/>' +
		'Pour les formats standards JPEG et PNG, il faudra trouver le bon taux de compression pour ne pas dégrader l’expérience de l’utilisateur.' +
		'Une bonne alternative peut être de les remplacer par des images au format WEBP malgré le fait que ce format ne soit pas encore supporté par tout les navigateurs. <br/><br/>' +
		'La taille est également importante. Pour économiser encore plus d’espace, les images doivent être rognées aux bonnes dimensions <br/>' +
		'Le format vectoriel SVG est nettement plus léger\xa0: il doit être privilégié pour chaque image graphique compatible avec ce format.'
	},
	{
		title: 'Catégorie 4\xa0: les vidéos',
		description: 'La première chose à faire avant d’ajouter une vidéo est de se demander si elle est vraiment nécessaire. ' + 
		'En effet, les videos demandent beaucoup de ressources\xa0: elles ne doivent donc pas être ajoutées juste pour illustrer ou pour remplir de l’espace. ' +
		'Dans le cas où elles sont nécessaires, l’autoplay doit être désactivé, surtout s’il y a du son dans la vidéo. <br/>' +
		'Vous devez également mettre en place le chargement différé sur chaque vidéo. <br/><br/>' +
		'Google recommande l’utilisation de vidéos plutôt que de GIF.'
	},
	{
		title: 'Catégorie 5\xa0: les animations',
		description: 'Comme pour les vidéos, la première chose concernant les animations est de se demander elles sont vraiment nécessaires\xa0: servent-elles le design ou sont-elles seulement décoratives ? <br/>' +
		'Si elles n’ont pas de réel intérêt, il vaut mieux ne pas les ajouter. <br/>' +
		'Le JavaScript est une technologie énergivore, il est donc important de la réserver aux tâches qui nécessitent vraiment de la logique. <br/><br/>' + 
		'Autant que possible, il vaudra mieux privilégier les effets en SCSS plutôt qu’en Javascript.'
	},
	{
		title: 'Catégorie 6\xa0: les librairies',
		description: 'L’accumulation de librairies est une cause récurrente de ralentissement des applications web, et donc de consommation excessive d’énergie. <br/>' +
		'Le design system est basé sur le framework Vuetify\xa0: il embarque donc la plupart des outils qui seront nécessaires au developpement de composant. <br/>' +
		'Si toutefois vous ne trouvez pas ce dont vous avez besoin dans les librairies présentes dans le projet, l’installation d’une nouvelle librairie doit se faire avec beaucoup de prudence. <br/>' +
		'Il faudra trouver la librairie la plus légère et adaptée possible, l’outils [bundlephobia](https://bundlephobia.com) permet rapidement d’évaluer la taille d’un package.'
	},
	{
		title: 'Catégorie 7\xa0: les fonctionnalités',
		description: 'Comme toujours, la première question à se poser est\xa0: est-ce que ma fonction est vraiment nécessaire ? <br/>' + 
		'Ne fait-elle pas doublon avec une fonction déjà existante ? <br/>' +
		'Dans le cas où elle est bien nécessaire, votre nouvelle fonctionnalité doit faire une chose précise. Moins elle enclanchera d’action, moins elle réclamera d\énergie.' +
		'Il faudra aussi faire attention à la façon de développer la fonction\xa0: les boucles inutiles ou les requêtes intempestives réclament là encore de mobiliser beaucoup de ressources. Il faut donc veiller à le faire à bon escient.'
	}
];