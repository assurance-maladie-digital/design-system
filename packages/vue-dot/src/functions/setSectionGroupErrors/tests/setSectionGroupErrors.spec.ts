import { incorrectFormErrors, questionErrors, sectionGroup } from './data/SectionGroupErrors';

import { setSectionGroupErrors } from '..';

// Tests
describe('setFormErrors', () => {
	it('sets error messages on different fields', () => {
		expect(setSectionGroupErrors(questionErrors, sectionGroup)).toMatchSnapshot();
	});

	it('doesn\'t sets an error on unknown field', () => {
		expect(setSectionGroupErrors(incorrectFormErrors, sectionGroup)).toMatchSnapshot();
	});
});
