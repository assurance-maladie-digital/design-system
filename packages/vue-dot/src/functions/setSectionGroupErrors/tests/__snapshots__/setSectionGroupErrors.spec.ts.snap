// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setFormErrors doesn't sets an error on unknown field 1`] = `
Object {
  "section1": Object {
    "questions": Object {
      "questionString": Object {
        "type": "text",
      },
    },
  },
  "section2": Object {
    "questions": Object {
      "questionPeriod": Object {
        "metadata": Object {
          "notExist": Object {
            "errorMessages": "Sub field does not exist",
          },
        },
        "type": "period",
      },
    },
  },
}
`;

exports[`setFormErrors sets error messages on different fields 1`] = `
Object {
  "section1": Object {
    "questions": Object {
      "questionString": Object {
        "metadata": Object {
          "errorMessages": Array [
            "error string 1",
            "error string 2",
          ],
        },
        "type": "text",
      },
    },
  },
  "section2": Object {
    "questions": Object {
      "questionPeriod": Object {
        "metadata": Object {
          "from": Object {
            "errorMessages": Array [
              "Error questionPeriod 1",
              "Error questionPeriod 2",
            ],
          },
          "to": Object {
            "errorMessages": Array [
              "Error questionPeriod to 1",
              "Error questionPeriod to 2",
            ],
          },
        },
        "type": "period",
      },
    },
  },
}
`;
